### ISSUE TITLE
Rqlite is returning 'NOT NULL constraint failed' even though the values are present

### ISSUE THREAD

**Issue by wellescastro** [rqlite/rqlite#1177]
**What version are you running?**
7.13.2
**Are you using Docker or Kubernetes to run your system?**
No
**Are you running a single node or a cluster?**
Single node
**What did you do?**
Given a rqlite database with the following schema:
"CREATE TABLE [TestTable] ([Id] varchar not null primary key, [Col1] varchar(10000) not null, [Col2] varchar, [Col3] varchar(500) not null, [Col4] varchar(15) not null, [Col5] varchar(100) not null, [Col6] varchar, [Col7] int not null, [Col8] int not null, [Col9] varchar(500))"

I repeatedly sent the following post request to the rqlite:
[
	["insert into TestTable (Id, Col1, Col2, Col3, Col4, Col5, Col6, Col7, Col8, Col9) values (:Id, :Val1, :Val2, :Val3, :Val4, :Val5, :Val6, :Val7, :Val8, :Val9)", {"Id":"62b4005b-4a0c-47f9-bcce-3600e893c71f","Val1":"Log.Test","Val2":"Test2","Val3":"Test3","Val4":"20230315161952","Val5":"Log.Test","Val6":"","Val7":0,"Val8":1,"Val9":null}]
]

**What did you expect to happen? **What happened instead?****
Instead of always getting a response with "error": "UNIQUE constraint failed: TestTable.Id" after the first insertion, it will give me random results including "NOT NULL constraint failed: TestTable.Col5", "NOT NULL constraint failed: TestTable.Col3", "NOT NULL constraint failed: TestTable.Col7", even though the values are present for the respective columns.
It seems that using a null value for a named parameter causes this strange behavior.

**Comment by otoolep** [comment]
Thank you for your report, I've reproduced the issue -- let me look into it.

**Comment by otoolep** [comment]
Reproduced with a simpler test case.
```bash
curl -XPOST 'localhost:4001/db/execute?pretty&timings' -H "Content-Type: application/json" -d '[
    "CREATE TABLE [TestTable] ([Id] int primary key, [Col1] int not null, [Col2] varchar(500), [Col3] int not null, [Col4] varchar(500))"
]'


curl -XPOST 'localhost:4001/db/execute?pretty&timings' -H "Content-Type: application/json" -d '
[
["insert into TestTable (Col1, Col2, Col3, Col4) values (:Val1, :Val2, :Val3, :Val4)", {
		"Val1": 1,
		"Val2": null,
		"Val3": 1,
		"Val4": null
	}]
]
'
```

**Comment by otoolep** [comment]
Most `INSERT`s went fine, then I got:
```json
{
    "results": [
        {
            "error": "NOT NULL constraint failed: TestTable.Col3"
        }
    ],
    "time": 0.010672685
}
```

**Comment by otoolep** [comment]
@wellescastro -- thanks again for your bug report, this was a bad one, surprised folks didn't hit it before (or were silently :-( )

Fixed release coming ASAP.

**Comment by wellescastro** [comment]
You're welcome @otoolep. I appreciate your prompt action on this issue :)

**Comment by otoolep** [comment]
Fixed on master.

**Comment by otoolep** [comment]
Fixed in 7.14.1.
